app/scripts/controllers/app-state.js

```diff
@@ -401,6 +405,29 @@ export default class AppStateController extends EventEmitter {
     this.store.updateState({ showAccountBanner });
   }
 
+  /**
+   * Sets an object with networkName and appName
+   * or `null` if the message is meant to be cleared
+   *
+   * @param switchedNetworkDetails
+   */
+  setSwitchedNetworkDetails(switchedNetworkDetails) {
+    this.store.updateState({ switchedNetworkDetails });
+  }
+
+  /**
+   * Remembers if the user prefers to never see the
+   * network switched message again
+   *
+   * @param switchedNetworkNeverShowMessage
```

Comment:
Nit: Here's the JSDoc type

```suggestion
   * @param {boolean} switchedNetworkNeverShowMessage
```

Reply:
Done!